specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkSecurityPerimeter.json

```diff
@@ -788,6 +940,165 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/networkSecurityPerimeters/{networkSecurityPerimeterName}/profiles/{networkSecurityPerimeterProfile}/checkMembers": {
+      "parameters": [
+        {
+          "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+        },
+        {
+          "$ref": "#/parameters/ResourceGroupNameParameter"
+        },
+        {
+          "$ref": "#/parameters/NetworkSecurityPerimeterNameParameter"
+        },
+        {
+          "$ref": "./network.json#/parameters/ApiVersionParameter"
+        },
+        {
+          "$ref": "#/parameters/NetworkSecurityPerimeterProfileParameter"
+        }
+      ],
+      "post": {
+        "tags": [
+          "NetworkSecurityPerimetersCheckMembers"
+        ],
+        "operationId": "NetworkSecurityPerimeters_CheckMembers",
+        "description": "Action to check members associated to an NSP Profile",
+        "parameters": [
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CheckMembers"
+            },
+            "description": "Parameter supplied to create the network security perimeter check members."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Check members, nothing is getting created or updated.",
+            "schema": {
+              "$ref": "#/definitions/CheckMembers"
```

Comment:
>CheckMembers [](http://example.com/codeflow?start=37&length=12)

POST is typically  an action on a resource. Why is the same object being returned as from the request body ? Typically we see some kind of status object for the action being returned

Reply:
The response contains similar data with a few more added keys so we are reusing the object.

Spec: https://msazure.visualstudio.com/One/_git/Networking-MNC?path=/doc/Design-Docs/Network-Security-Perimeter-Design-Specification-Annex.md&version=GBmaster&_a=preview&anchor=request

Reply:
How are members created? Is this created by your RP on behalf of a user? if yes, then this is a proxy resource, and I suggest creating a new resource type: `members` where it returns a collection get, also a single element, either: `members/default` (singleton) or `members/{memberName}` your API is effectively returning data of some sort instead of executing an action on your resource (e.g. shutdown, reboot, etc.)

Reply:
One more comment, how long does it takes to return members data? Is it stored in different data stores or is a quick lookup?

Reply:
We need to send multiple different items in an array so we are using POST. This is an internal API.

According to ARM team we should POST for this API : https://stackoverflow.microsoft.com/questions/258519

 Is it stored in different data stores or is a quick lookup? : It's a quick lookup.